backend/algorithms/tests/test_user_has_taken.py

```diff
@@ -0,0 +1,80 @@
+"""
+Tests `User.has_taken_course`
+    - Given a list of courses, return if the course has been taken (bool)
+    - Want checks for course fail and pass
+"""
+
+import json
+from algorithms.objects.user import User
+
+EXAMPLE_USERS_PATH = "./algorithms/tests/exampleUsers.json"
+
+with open(EXAMPLE_USERS_PATH) as f:
+    USERS = json.load(f)
+
+
+def test_user_has_taken_empty_user():
+    """
+    The user object being passed in for construction is empty.
+    Note that this is differentiated from the case where the user
+    has no courses.
+    """
+    user = User()
+    assert user.has_taken_course("COMP1069") is False
```

Comment:
i think i remember when these were getting written the rationale was

- Using `assert not user.has_taken_course`=
    - Pros: very pythonic
- Using `assert user.has_take_course is False`
    - mmm lines have very pretty aligned aesthetic even when you assert for the true case

Reply:
lmao nice. There is the semantic difference that has_taken_course must also be a boolean, but we have mypy for that now 😎